low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -252,13 +301,25 @@ def connect_to_prop_changed_signal(self, service):
             logger.debug(f"{service}({pid}) state is {state}:{substate}")
 
             if state in  ["activating", "reloading", "deactivating"]:
-                self.not_active_services[service] = \
-                                    [self.current_time(), "N/A", "N/A"]
-            elif state != "active":
+                if service not in self.not_active_services:
+                    self.not_active_services[service] = \
+                                    [self.current_time(), state, substate]
+            elif state != "active" and service not in self.failed_services:
                 self.failed_services.append(service)
-                self.raise_alert(service, "N/A", state, "N/A", substate,
-                                    "N/A", pid, 0)
+                self.raise_alert(service, prev_state, state, prev_substate,
+                                 substate, prev_pid, pid, 0)
                 logger.error(f"{service} is not active initially. state = {state}:{substate}")
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
```

Comment:
addressed.

Reply:
I reverted it since it was putting the wrong info in the persistent cache if the list is updated after the alert is sent.
Even if the alert is not sent, we are handling it using accumulated state right? I don't think this will cause any issues.

Reply:
I Think it is because raise_alert function is doing multiple things

1. Send alert to egress
2. Update cache

raise_alert should only do one thing - Send alert to egress
You can create separate function(extract function from raise_alert) for updating cache. Then, do things in below order

1. raise_alert
2. failed_services.remove(service)
3. update_cache

Reply:
Yes @sandeepsgt , we can you that in this function but if you look at the `action_per_transition` function then at each place where I am setting alert index we would have to call `raise_alert` and then `emove service from `failed_servies` or `non_active_services` list and then update the cache. That would make the code complicated.

Reply:
updated the flow to :
`raise_alert    -->    failed_service/not_active_services.remove(service)    -->    update_persistent_cache.`

except in action_per_transition()  where the flow is :
 `failed_service/not_active_services.remove(service)      -->     raise_alert     -->     update_persistent_cache.`